mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -43,6 +50,187 @@ struct CompactBlockRow {
     data: Vec<u8>,
 }
 
+pub type OnCompactBlockFn<'a> =
+    dyn FnMut(TonicCompactBlock) -> Result<(), MmError<UpdateBlocksCacheErr>> + Send + Sync + 'a;
+
+#[async_trait]
+pub trait ZRpcOps {
+    async fn get_block_height(&mut self) -> Result<u64, MmError<UpdateBlocksCacheErr>>;
+
+    async fn scan_blocks(
+        &mut self,
+        start_block: u64,
+        last_block: u64,
+        on_block: &mut OnCompactBlockFn,
+    ) -> Result<(), MmError<UpdateBlocksCacheErr>>;
+
+    async fn check_watch_for_tx(&mut self, tx_id: TxId) -> Result<(), ()>;
```

Comment:
I think `check_tx_existence` will be a more suitable name here. And it should either return bool or have a more specific return type instead of just `Result<(), ()>`.